sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/catalog/SessionCatalog.scala

```diff
@@ -523,13 +524,14 @@ class SessionCatalog(
    */
   def dropPartitions(
       tableName: TableIdentifier,
-      parts: Seq[TablePartitionSpec],
+      specs: Seq[TablePartitionSpec],
       ignoreIfNotExists: Boolean): Unit = {
+    requirePartialMatchedPartitionSpec(specs, getTableMetadata(tableName))
```

Comment:
Wait, I thought the intention of this patch is to ban the partial matching, which is buggy in Hive? Looks like this just ensures that the user doesn't specify a column that is not a partitioned column here. Is that sufficient to bypass the bug in Hive that you were talking about?